torchaudio/prototype/functional/_dsp.py

```diff
@@ -78,3 +79,96 @@ def oscillator_bank(
     if reduction == "mean":
         return waveform.mean(-1)
     return waveform
+
+
+def adsr_envelope(
+    num_frames: int,
+    *,
+    attack: float = 0.0,
+    hold: float = 0.0,
+    decay: float = 0.0,
+    sustain: float = 1.0,
+    release: float = 0.0,
+    n_decay: int = 2,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+):
+    """Generate ADSR Envelope
+
+    .. devices:: CPU CUDA
+
+    Args:
+        num_frames (int): The number of output frames.
+        attack (float): The relative *time* it takes to reach the maximum level from the start.
```

Comment:
similarly w/ other params
```suggestion
        attack (float, optional): The relative *time* it takes to reach the maximum level from the start. (Default: 0.0)
```